text/0000-transaction-payload/0000-transaction-payload.md

```diff
@@ -0,0 +1,562 @@
+The new transaction structure should fulfil the following criteria:
+* Support for Ed25519 (and thus reusable addresses).
+* Support for adding new types of signature schemes, addresses, inputs, and outputs as part of protocol upgrades.
+* Self-contained, as in being able to validate the transaction immediately after receiving it.
+* Enable unspent transaction outputs (UTXO) as inputs instead of an account based model (UTXO enables easier double-spend detection).
+
+# Detailed design
+
+## UTXO
+
+The *unspent transaction output* (UTXO) model defines a ledger state where balances are not directly associated to addresses but to the outputs of transactions. In this model, transactions specify the outputs of previous transactions as inputs, which are consumed to create new outputs. A transaction must consume the entirety of the specified inputs.
+
+Using an UTXO based model provides several benefits:
+* Parallel validation of transactions.
+* Easier double-spend detection, since conflicting transactions would reference the same UTXO.
+* Replay-protection which is important when having reusable addresses. Replaying the same transaction would manifest itself as already being applied or existent and thus not have any impact.
+* Technically seen, balances may no longer be associated to addresses which raises the level of abstraction and thus enables other types of outputs. Consider, for example, a type of output which specifies the balance to be unlocked by a transaction which must fulfil a Proof-of-Work difficulty or supply some other unlock criteria, etc.
+
+Within a transaction using UTXOs, inputs and outputs make up the to-be-signed data of the transaction. The section unlocking the inputs is called *unlock block*. An unlock block may contain a signature proving ownership of a given input's address and/or other unlock criteria.
+
+The following image depicts the flow of funds using UTXO:
+
+![UTXO flow](https://i.imgur.com/h3uxf6N.png)
+
+The way UTXOs are referenced is further described in the <i>Structure</i> section of this RFC.
+
+## Structure
+
+### Serialized Layout
+
+A <i>Transaction Payload</i> payload is made up of two parts:
+1. The <i>The Transaction Essence</i> part which contains the inputs, outputs and an optional embedded payload.
+2. The <i>Unlock Blocks</i> which unlock the <i>Transaction Essence</i>'s inputs. In case the unlock block contains a signature, it signs the entire <i>Transaction Essence</i> part.
+
+All values are serialized in little-endian encoding. The serialized form of the transaction is deterministic, meaning the same logical transaction always results in the same serialized byte sequence.
+
+A `Blake2b-256` hash of the entire serialized data makes up <i>Transaction Payload</i>'s ID.
+
+Following table structure describes the entirety of a <i>Transaction Payload</i>'s serialized form:
+* [Data Type Notation](https://github.com/GalRogozinski/protocol-rfcs/blob/message/text/0017-message/0017-message.md#data-types)
```

Comment:
This seems to be referenced in several places, would it make sense to have an RFC to define the datatypes which would be the canonical location instead of RFCs cross linking each other?

Reply:
I agree. However since this effects several RFCs it should happen in a separate RFC. I'll create an issue for that.